mountpoint-s3/src/fs.rs

```diff
@@ -40,6 +40,7 @@ struct FileHandle<Client: ObjectClient, Runtime> {
     inode: Inode,
     full_key: String,
     object_size: u64,
+    etag: Option<String>,
```

Comment:
Probably put this into the `FileHandleType::Read` enum variant instead of here, since it only makes sense for reads. And then once we do that: can the ETag for an open read handle ever be `None`? If not, that would make us feel a lot more confident that we haven't forgotten an ETag check somewhere—the type system says we'll always have one if we don't make this an `Option`.